src/Utils.ps1

```diff
@@ -84,7 +84,8 @@ function Add-PoshGitToProfile([switch]$AllHosts, [switch]$Force, [switch]$WhatIf
     $profilePath = if ($AllHosts) { $PROFILE.CurrentUserAllHosts } else { $PROFILE.CurrentUserCurrentHost }
 
     # Under test, we override some variables using $args as a backdoor.
-    if ($args.Count -gt 0) {
+    # TODO: Can we just turn these into optional parameters with well-defined behavior?

```

Comment:
I'd rather not make the public parameters any more complicated for the sake of tests.  Yes, we could use env vars but those are global to the process and could impact tests if they ever ran in parallel (which they currently don't).